Sources/SourceControl/GitRepository.swift

```diff
@@ -81,12 +81,26 @@ public struct GitRepositoryProvider: RepositoryProvider {
         // expected cost of iterative updates on a full clone is less than on a
         // shallow clone.
         precondition(!localFileSystem.exists(path))
+
         // FIXME: Ideally we should pass `--progress` here and report status regularly.  We currently don't have callbacks for that.
         try self.callGit("clone", "--mirror", repository.url, path.pathString,
                          repository: repository,
                          failureMessage: "Failed to clone repository \(repository.url)")
     }
-
+    
+    public func isInitializedDirectory(location: String) throws {
+        // Provides better feedback when mistakingly using url: for a dependency that
+        // is a local package. Still allows for using url with a local package that has
+        // also been initialized by git
+        do {
+            try self.callGit("-C", location, "rev-parse", "--git-dir", repository: RepositorySpecifier(url: location))
+        } catch let error as GitCloneError {
+            throw GitCloneError(repository: RepositorySpecifier(url: location),
+                                message: "Cannot clone from local directory \(location)\nPlease git init or use \"path:\" for \(location)",
```

Comment:
this message is to specific. imo this API should simply return a boolean (isValidDirectory) and the calling code can throw a message of this type if it is returning false

Reply:
i.e. 

```
public func isValidDirectory(_ directory: String) -> Bool {
   ...
   return true
} catch let error as GitCloneError {
  return false
}
``